### DISCUSSION TITLE
API / WS

### DISCUSSION THREAD

**Discussion by MarcJenningsUK** [NRCHKB/node-red-contrib-unifi-os#21] [General]
Hey.  I've been poking around in pyunifiprotect and found in the file
https://github.com/briis/pyunifiprotect/blob/master/pyunifiprotect/unifi_protect_server.py
lines 183 and 184 there are a couple of potential endpoints available
  self.api_path = "proxy/protect/api"            
  self.ws_path = "proxy/protect/ws"

I have no idea if you'd seen those, but that python library is able to get doorbell messages.  I found it by looking at the HomeAssistant plugin for unifi protect at https://github.com/briis/unifiprotect

**Comment by crxporter** [comment]
If you look at that link, the magic part we don't have (yet) is in line 1073. The addition of last update at the end of the `/proxy/protect/ws/updates`... once @Shaquu is back in programming mode for this project I hope this part is on top of the list. 

It should allow doorbell events, motion events, and smart detect events to be pushed into NR!! That's the dream!

Anyway, yes. That is part of our goal here.
